editoast/src/views/infra/auto_fixes/operational_point.rs

```diff
@@ -1,26 +1,86 @@
+use editoast_schemas::primitives::OSRDIdentified;
 use editoast_schemas::primitives::OSRDObject as _;
 use editoast_schemas::primitives::ObjectRef;
+use editoast_schemas::primitives::ObjectType;
+use itertools::Itertools;
+use json_patch::Patch;
+use json_patch::PatchOperation;
+use json_patch::RemoveOperation;
 use std::collections::HashMap;
-
 use tracing::debug;
 
-use super::new_ref_fix_delete_pair;
 use super::Fix;
+use super::OrderedOperation;
 use crate::generated_data::infra_error::InfraError;
 use crate::generated_data::infra_error::InfraErrorType;
 use crate::infra_cache::object_cache::OperationalPointCache;
+use crate::infra_cache::operation::CacheOperation;
+use crate::infra_cache::operation::DeleteOperation;
+use crate::infra_cache::operation::Operation;
+use crate::infra_cache::operation::UpdateOperation;
+use crate::infra_cache::ObjectCache;
+
+fn invalid_reference_to_ordered_operation(
+    operational_point: &OperationalPointCache,
+    object_ref: &ObjectRef,
+) -> Option<OrderedOperation> {
+    let (track_range_idx, _) = operational_point
+        .parts
+        .iter()
+        .enumerate()
+        .find(|(_idx, part)| part.track.as_str() == object_ref.obj_id)?;
```

Comment:
Indeed, but we should have one error per invalid ref. So in the end, all the invalid refs will be fixed.

Reply:
The fix trial is applied only once looped on all errors, so we would have fix-duplicates (I tested it in this branch https://github.com/OpenRailAssociation/osrd/compare/fam/add-autofix-op...peb/add-autofix-op?expand=1).
And those duplicates would lead to unwanted behavior (removing wrong part in the test case I modified).

Then the larger loop on (process errors + compute fixes) would also save some of the duplicates I guess, but too late.

So if we want to be safe for this case, I think we have to at least deduplicate on reference, then probably here return a vector of OrderedOperations (letting the larger loop do the job is a bit hazardous IMO)

Reply:
I chose to deduplicate the final list of `OrderedOperations`. I also fix this in the `electrifications` and `speed_section` auto fixes.

Reply:
OK, fine by me.
Just one match (OP on the same unknown track) is fixed at each iteration, the larger loop is responsible for the possible others (and may fail returning nothing if there are too much of them).
Modified test from my branch behaves like that ✔️ 

Not worth the effort, but just sharing possible improvements:
* return a list of `OrderedOperations` in `invalid_reference_to_ordered_operation()` to solve all the matches in one iteration
* dedup before calling `invalid_reference_to_ordered_operation` to save calls
* introduce a dedup above all operations of all fixes of an error iteration (though one)